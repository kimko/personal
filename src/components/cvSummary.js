import React from "react";
import PropTypes from "prop-types";

export const Summary = (props) => {
  return (
    <article className="bb w-100 cf f6 pt5 pl4">
      <div className="flex flex-wrap">
        <div className="w-100 w-third-ns">
          <ShortDescription
            shortDescription={props.shortDescription}
            title={props.title}
          />
        </div>
        <div className="w-100 w-two-thirds-ns pl4-ns">
          <SummaryList summary={props.summary} />
        </div>
      </div>
    </article>
  );
};

Summary.propTypes = {
  shortDescription: PropTypes.string,
  title: PropTypes.string,
  summary: PropTypes.array,
};

const ShortDescription = (props) => {
  return (
    <article className="measure-narrow measure-ns pt2">
      <h1 className="bl bw2 f5 pl1">{props.title}</h1>

      <p className="">{props.shortDescription}</p>
    </article>
  );
};

ShortDescription.propTypes = {
  shortDescription: PropTypes.string,
  title: PropTypes.string,
};

const SummaryList = ({ summary }) => {
  return (
    <article>
      <ul className="pl3 pt4-ns pr2">
        {summary.map((item, idx) => (
          <li key={idx} className="pv1 mw7">
            {item}
          </li>
        ))}
      </ul>
    </article>
  );
};

SummaryList.propTypes = {
  summary: PropTypes.array,
};
