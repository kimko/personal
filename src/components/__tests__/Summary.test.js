import React from "react";
import { render, cleanup } from "@testing-library/react";

import { Summary } from "../cvSummary";

afterEach(cleanup);

const title = "title asdf";
const shortDescription = `description asdf`;
const summary = ["qwert", "yuiop", "asdf", "ghjkl", "zxcvb"];

it("renders a Summary", () => {
  const { getByTestId, asFragment } = render(
    <Summary
      title={title}
      shortDescription={shortDescription}
      summary={summary}
    />,
  );

  expect(getByTestId("test-article-summary")).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
