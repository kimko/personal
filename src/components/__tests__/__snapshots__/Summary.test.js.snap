// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Summary 1`] = `
<DocumentFragment>
  <article
    class="bb w-100 cf f6 pt5 pl4"
  >
    <div
      class="flex flex-wrap"
    >
      <div
        class="w-100 w-third-ns"
      >
        <article
          class="measure-narrow measure-ns pt2"
        >
          <h1
            class="bl bw2 f5 pl1"
          >
            title asdf
          </h1>
          <p
            class=""
          >
            description asdf
          </p>
        </article>
      </div>
      <div
        class="w-100 w-two-thirds-ns pl4-ns"
      >
        <article
          data-testid="test-article-summary"
        >
          <ul
            class="pl3 pt4-ns pr2"
          >
            <li
              class="pv1 mw7"
            >
              qwert
            </li>
            <li
              class="pv1 mw7"
            >
              yuiop
            </li>
            <li
              class="pv1 mw7"
            >
              asdf
            </li>
            <li
              class="pv1 mw7"
            >
              ghjkl
            </li>
            <li
              class="pv1 mw7"
            >
              zxcvb
            </li>
          </ul>
        </article>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
